tool/lib/test/unit.rb

```diff
@@ -882,6 +911,61 @@ def setup_options(opts, options)
         opts.on '--most-asserted=N', Integer, 'Show most asserted N tests' do |n|
           options[:most_asserted] = n
         end
+        opts.on '--launchable-test-reports=PATH', String, 'Report test results in Launchable JSON format' do |path|
+          options[:launchable_test_reports] = writer = JsonStreamWriter(path)
+          writer.write_array('testCases')
+          at_exit{ writer.close }
+        end
+      end
+      class JsonStreamWriter # :nodoc: all
+        def initialize(path)
+          @file = File.open(path, 'w')
+          @file.write('{')
+          write_new_line
+        end
+        def batch_write
+          @file.write("    ")
+          @file.write("{")
+          write_new_line
+          yield
+          @file.write("}")
+          write_comma
+          write_new_line
+        end
+        def write_array(key)
+          @file.write("  ")
+          @file.write(key)
```

Comment:
`IO#write` takes multiple arguments.
```suggestion
          @file.write("  ", key)
```